crates/optimism/rpc/src/sequencer.rs

```diff
@@ -46,54 +47,32 @@ impl SequencerClient {
         &self.inner.http_client
     }
 
-    /// Returns the next id for the request
-    fn next_request_id(&self) -> usize {
-        self.inner.id.fetch_add(1, atomic::Ordering::SeqCst)
-    }
-
-    /// Helper function to get body of the request with the given params array.
-    fn request_body(&self, method: &str, params: Value) -> serde_json::Result<String> {
-        let request = json!({
-            "jsonrpc": "2.0",
-            "method": method,
-            "params": params,
-            "id": self.next_request_id()
-        });
-
-        serde_json::to_string(&request)
-    }
-
-    /// Sends a POST request to the sequencer endpoint.
-    async fn post_request(&self, body: String) -> Result<(), reqwest::Error> {
-        self.http_client()
-            .post(self.endpoint())
-            .header(reqwest::header::CONTENT_TYPE, "application/json")
-            .body(body)
-            .send()
-            .await?;
+    /// Sends a [`alloy_rpc_client::RpcCall`] request to the sequencer endpoint.
+    async fn send_rpc_call(&self, method: &str, params: Value) -> Result<(), SequencerClientError> {
+        self.http_client().request::<Value, ()>(method.to_string(), params).await.inspect_err(
+            |err| {
+                warn!(
+                    target: "rpc::sequencer",
+                    %err,
+                    "HTTP request to sequencer failed",
+                );
+            },
+        )?;
         Ok(())
     }
 
     /// Forwards a transaction to the sequencer endpoint.
     pub async fn forward_raw_transaction(&self, tx: &[u8]) -> Result<(), SequencerClientError> {
-        let body = self
-            .request_body("eth_sendRawTransaction", json!([format!("0x{}", hex::encode(tx))]))
-            .map_err(|_| {
+        self.send_rpc_call("eth_sendRawTransaction", json!([format!("0x{}", hex::encode(tx))]))
```

Comment:
this then can just use a tuple `(encoded,)`

see also 

https://github.com/alloy-rs/alloy/blob/72a12eab59df688abce53da845ee66dcc1060c86/crates/provider/src/provider/trait.rs#L854-L854

Reply:
yups , makes sense, thanks for code reference, changed it.